### DISCUSSION TITLE
if (Boolean.parseBoolean(System.getProperty(SERIALIZATION_SECURITY_CHECK_KEY, "true")))cause some thread enter blocked

### DISCUSSION THREAD

**Discussion by HappyinessHome** [apache/dubbo#14945] [General]
### Pre-check

- [X] I am sure that all the content I provide is in English.

### Apache Dubbo Component

Java SDK (apache/dubbo)

### Details

"In the DecodeableRpcInvocation.decode method of Dubbo, the code:

if (Boolean.parseBoolean(System.getProperty(SERIALIZATION_SECURITY_CHECK_KEY, "true"))) {
    CodecSupport.checkSerialization(frameworkModel.getServiceRepository(), path, version, serializationType);
}
[Dubbo_JStack.log.2024-11-15_00_25_12.txt](https://github.com/user-attachments/files/17900432/Dubbo_JStack.log.2024-11-15_00_25_12.txt)

if (Boolean.parseBoolean(System.getProperty(SERIALIZATION_SECURITY_CHECK_KEY, "true"))) {
    CodecSupport.checkSerialization(frameworkModel.getServiceRepository(), path, version, serializationType);
}
calls System.getProperty(SERIALIZATION_SECURITY_CHECK_KEY), which leads to Hashtable operations causing some threads to enter a BLOCKED state. This results in thread exhaustion exceptions being thrown. However, I'm not sure if this is the root cause. Below is the content of dubbo_jstack.log. Please help me. The Dubbo version is 2.7.2, and the JDK version is 8."

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by AlbumenJ** [comment]
I would recommend you to upgrade your Dubbo version first.
